src/main/java/org/owasp/esapi/logging/appender/LogPrefixAppender.java

```diff
@@ -35,6 +39,18 @@ public class LogPrefixAppender implements LogAppender {
     private final boolean logApplicationName;
     /** Application Name to record. */
     private final String appName;
+    /** Whether to omit event type in logs or not. */
+    private static boolean omitEventTypeInLogs;
+
+    static {
+
+        try {
+            omitEventTypeInLogs =
+                    ESAPI.securityConfiguration().getBooleanProp(OMIT_EVENT_TYPE_IN_LOGS);
+        } catch (ConfigurationException ex) {
+            omitEventTypeInLogs = false;
+        }
```

Comment:
ConfigurationException should not be caught.

Additionally, doing this in the static block may prevent a misconfigured application from correctly propagating the runtime exception. 

I realize that the static keyword is intended to optimize the number of instances of this variable, but this project really does have enough static content.  Let the variable die with the instance.

Please consider:
1.  Removing the catch from the initialization -- that exception should escape if it happens (See DefaultSecurityProvider.getBooleanProp)
2. make the omitEventTypeInLogs a member varable of the class (remove static modifier), and move the initialization to the constructor.

Reply:
I see @kwwall  and I are in disagreement about the ConfigurationException.   Please disregard my  suggestion and follow his guidance.

I believe moving the initialization to the constructor and making it non-static is still valid.

Apologies for not being more dilligent about reading the previous conversation before jumping into the code.

Reply:
My reason for making it static and putting it in a static initializer goes as follows:

1. There are potentially a LOT of uses of ESAPI logging statements so the `LogPrefixAppender` CTOR will be called many times.
2. It seems likely that many applications will not have the new `Logger.OmitEventTypeInLogs` property.
3. If that property is missing, as previously discussed, it will cause a `ConfigurationException` to be thrown each time.
4. Throwing / catching exceptions is a relatively expensive operation from a Java execution perspective.
5. Overall, this could make the effect of logging considerably slower.
6. If we make it a class variable instead and place it in a static initializer as I suggested, we only suffer that throw/catch overhead once.
7. We generally assume that the value of an ESAPI property does not change in the course of a running application, so given that assumption, this is a sound, although rather unorthodox approach.

Reply:
I appreciate the explanation and understand the intention.  Thank you for the consideration.

Resolving conversation, no changes requested/expected.